### DISCUSSION TITLE
충돌했을 때 오브젝트의 색상을 바꾸고싶습니다

### DISCUSSION THREAD

**Discussion by yoon-hw22** [naverz/zepeto-studio-kor#967] [5. Q&A]
유니티 자체에서는 간단한 코드인데 제페토로 구현을 하려고하니 어렵네요 ㅠㅠ 
어떻게 작성해야하나요? 캐릭터와 큐브가 충돌했을 때 색상을 변경하고싶습니다. 

import { GameObject,Input,Renderer,Collider,Vector3,Quaternion, Material, Cubemap, Color } from 'UnityEngine';
import { ZepetoCharacter, ZepetoPlayer, ZepetoPlayers } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script';
import { SceneManager } from 'UnityEngine.SceneManagement';
import { UxmlValueBounds } from 'UnityEngine.UIElements';

export default class ColorChange extends ZepetoScriptBehaviour 
{
    //제페토 캐릭터를 인식
    
    private zepetoCharacter: ZepetoCharacter;
    
    
    Start() 
    {    
        //Zepeto character object 인식
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.zepetoCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
        }); 
        
        

    }    
    
    OnCollisionEnter(collider: Collider) 
    {
        //캐릭터를 제대로 못 불러온 경우 return 
        if (this.zepetoCharacter == null || collider.gameObject != this.zepetoCharacter.gameObject) 
            return;
        
        this.GetComponent<Renderer>().material.color = Color.blue;
        
       
    }
}

**Comment by YSHFM** [comment]
https://www.mettit.com/pt/view/PRE20220512190033673

참고할만한 링크 올려둡니다

쉐이더를 이용하여 해당 함수를 이용해 컬러값을 변경할 수도 있습니다.

 this.gameobject.GetComponent<Renderer>().material.SetColor(propertyname,color);
